src/assets/JourneyWithGo-ABlogSeries/goControlFlow.md

```diff
@@ -0,0 +1,378 @@
+---
+title: Control Statements - Control Your Go
+authorName: Prarup Gurung (प्रारुप गुरुङ्ग)
+authorAvatar: https://www.jankaritech.com/images/2021/12/29/p1110938-crop-colour-800.jpg
+authorLink: https://github.com/grgprarup
+createdAt: Dec 13, 2022
+tags: branching statement, if-else, switch, looping statement, control flow statement
+banner: https://camo.githubusercontent.com/2b507540e2681c1a25698f246b9dca69c30548ed66a7323075b0224cbb1bf058/68747470733a2f2f676f6c616e672e6f72672f646f632f676f706865722f6669766579656172732e6a7067
+seriesTitle: Journey With Go - A Blog Series
+episode: 2
+---
+
+**[गो](https://go.dev "Go in Nepali")** **"Journey With Go - A Blog Series"** about the basics of Go `Go Basic`, advanced concepts in Go `Go Beyond Basics`, testing with go `Godog as Test Framework` and many more.
+
+This is the second installment of the series. You are here that means you have already read the first blog of the series. If you haven't read the first blog, please read it first. It is available [here](https://blog.jankaritech.com/#/blog/Journey%20With%20Go%20-%20A%20Blog%20Series/Go%20Basics%20-%20The%20Starting%20Point "here"). In this blog, we will learn about the `Control Flow Statements in Go`. **So, let's get started.**
+
+## **Control Flow Statements**
+
+These statements are used to control the flow of the program. In other words, they control the execution of the program. There are two types of control flow statements in Go.
+
+1. Branching Statements
+2. Looping Statements
+
+### **A. Branching Statements**
+
+Branching statements divides the program into different branches on the basis of given condition. Some blocks of code might execute and some might not. In Go, we have `if-else` and `switch` to control the flow of program.
+
+#### 1. if-else
+
+If the condition is true, it executes the code inside the `if` block otherwise (i.e. the condition is false), it executes the code inside the `else` block. There are different forms of `if-else` pairs such as, `if`, `if-else`, `nested if-else`, and `else-if ladder`.
+
+##### _a. if_
+
+This can be use if you want to execute the code only if the condition is true. The code inside of the `if` block gets executed if the condition is true. The syntax of `if` is as follows.
+
+```go
+if condition {
+    // code to execute if the condition is true
+}
+```
+
+Example:
+
+![image](/src/assets/JourneyWithGo-ABlogSeries/images/if-example.png "If Example")
+
+Output:
+
+```go
+You are eligible to receive a Citizenship
+```
+
+##### _b. if-else_
+
+When you want to execute codes either for true or false case then `if-else` can be used. `if-else` executes the code inside of the `if` block if the condition is true otherwise, it executes the code inside of the `else` block. The syntax of `if-else` is as follows.
```

Comment:
```suggestion
When you want to execute codes either for true or false cases then `if-else` can be used. `if-else` executes the code inside of the `if` block if the condition is true otherwise, it executes the code inside of the `else` block. The syntax of `if-else` is as follows.
```